src/pages/data/[[...slug]].js

```diff
@@ -62,6 +64,24 @@ export async function getStaticProps({ params, preview, previewData }) {
   }
 
   const blocks = formatBlocksForSections(props?.post?.blocks);
+  blocks.documentsAndDatasets =
+    (await Promise.all(
+      blocks?.documentsAndDatasets?.map(
+        async ({ type, items: originalItems, ...others }) => {
+          let items = originalItems;
+          if (type === "datasets") {
+            items = await Promise.all(items.map(fetchOpenAfricaDatasets));
+          } else if (type === "documents") {
+            items = await Promise.all(items.map(fetchSourceAfricaDocuments));
+          }
+          // A single url can contain multiple datasets/documents & hence the
+          // need for flat(1)
+          items = items.flat(1);
+
+          return { ...others, type, items };
+        }
+      )
+    )) || null;
```

Comment:
@kilemensi doesn't this belong to `formatBlocksForSections` more?

Reply:
Probably @KobbyMmo ... `formatBlocksForSections` just scares me!

Reply:
aww, the old man is afraid of bytes. 😂😂